[BUG] BUG-01: Compile error in exec_in — type mismatch on lo3_val

## Summary

`exec_in` does not compile due to two type mismatches. The variable `temp` is of type `lo3_val` but is passed directly where `int` and `char *` are expected. This means the `#i` (stdin) instruction is **completely non-functional** — the project cannot build at all in its current state.

## Affected file

`src/execute.c`, lines 266 and 268

```c
lo3_val temp = pars_resv(buf);

// ...
var_setNum(name, temp);    // ❌ Error: lo3_val passed, int expected
var_setString(name, temp); // ❌ Error: lo3_val passed, char * expected
```

## Compiler output

```
src/execute.c:266:34: error: incompatible type for argument 2 of 'var_setNum'
src/execute.c:268:37: error: incompatible type for argument 2 of 'var_setString'
```

## Suggested fix

Extract the correct field from the `lo3_val` union before passing it:

```c
if (!a2.chooseType) {
    var_setNum(name, temp.value.num);
} else {
    var_setString(name, temp.value.string);
}
```

cc @seesee010

---
*This issue was identified and created by [Claude](https://claude.ai) on behalf of the repository maintainer.*

--- a/src/execute.c
+++ b/src/execute.c
@@ -268,8 +268,8 @@ void exec_in(lo3_val a1, lo3_val a2, char array[2]) {
 	}
 
 	if (!a2.chooseType) {
-		var_setNum(name, temp);
+		var_setNum(name, temp.value.num);
 	} else {
-		var_setString(name, temp);
+		var_setString(name, temp.value.string);
 	}
 }

